src/main/java/baritone/process/ElytraProcess.java

```diff
@@ -329,4 +351,59 @@ public double placeBucketCost() {
             return COST_INF;
         }
     }
+
+    private static boolean isInBounds(BlockPos pos) {
+        return pos.getY() >= 0 && pos.getY() < 128;
+    }
+
+    private boolean isAtEdge(BlockPos pos) {
+        return ctx.world().isAirBlock(pos.north())
+                || ctx.world().isAirBlock(pos.south())
+                || ctx.world().isAirBlock(pos.east())
+                || ctx.world().isAirBlock(pos.west())
+                // corners
+                || ctx.world().isAirBlock(pos.north().west())
+                || ctx.world().isAirBlock(pos.north().east())
+                || ctx.world().isAirBlock(pos.south().west())
+                || ctx.world().isAirBlock(pos.south().east());
+    }
+
+    private boolean isSafeLandingSpot(BlockPos pos) {
+        BlockPos.MutableBlockPos mut = new BlockPos.MutableBlockPos(pos);
+        while (mut.getY() >= 0) {
+            IBlockState state = ctx.world().getBlockState(mut);
+            Block block = state.getBlock();
+
+            if (block == Blocks.NETHERRACK || block == Blocks.GRAVEL || block == Blocks.NETHER_BRICK) {
+                return true;
+            } else if (block != Blocks.AIR) {
+                return false;
+            }
+            mut.setPos(mut.getX(), mut.getY() - 1, mut.getZ());
+        }
+        return false; // void
+    }
+
+    private BetterBlockPos findSafeLandingSpot() {
+        final BetterBlockPos start = ctx.playerFeet();
+        Queue<BetterBlockPos> queue = new LinkedList<>();
```

Comment:
```suggestion
        Queue<BetterBlockPos> queue = new PriorityQueue(Comparator.comparingInt(pos -> pos.getY()).reversed());
```

maybe. solid maybe. definite maybe. definitely test this before doing it.